pkg/executor/distsql.go

```diff
@@ -592,17 +596,24 @@ func (e *IndexLookUpExecutor) buildTableKeyRanges() (err error) {
 	} else {
 		physicalID := getPhysicalTableID(e.table)
 		var kvRanges *kv.KeyRanges
-		if e.index.ID == -1 {
-			kvRanges, err = distsql.CommonHandleRangesToKVRanges(dctx, []int64{physicalID}, e.ranges)
+		if e.index.IsFulltextIndex() {
+			kvRanges, err = distsql.FulltextIndexRangesToKVRanges(dctx, []int64{physicalID}, e.ranges)
 		} else {
-			kvRanges, err = distsql.IndexRangesToKVRangesWithInterruptSignal(dctx, physicalID, e.index.ID, e.ranges, e.memTracker, nil)
+			if e.index.ID == -1 {
+				kvRanges, err = distsql.CommonHandleRangesToKVRanges(dctx, []int64{physicalID}, e.ranges)
+			} else {
+				kvRanges, err = distsql.IndexRangesToKVRangesWithInterruptSignal(dctx, physicalID, e.index.ID, e.ranges, e.memTracker, nil)
+			}
 		}
 		e.kvRanges = kvRanges.FirstPartitionRange()
 	}
 	return err
 }
 
 func (e *IndexLookUpExecutor) open(_ context.Context) error {
+	if e.storeType == kv.TiFlash && !e.batchCop {
```

Comment:
I think we can hard code here to always use `batchcop` if e.storeType is TiFlash?

Reply:
addressed.